infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -67,15 +72,35 @@
      * @param serverContext Initialized Server Context.
      */
     public CorfuServerNode(@Nonnull ServerContext serverContext) {
-        this(serverContext,
-                ImmutableMap.<Class, AbstractServer>builder()
-                        .put(BaseServer.class, new BaseServer(serverContext))
-                        .put(SequencerServer.class, new SequencerServer(serverContext))
-                        .put(LayoutServer.class, new LayoutServer(serverContext))
-                        .put(LogUnitServer.class, new LogUnitServer(serverContext))
-                        .put(ManagementServer.class, new ManagementServer(serverContext))
-                        .build()
-        );
+
+        LogUnitServerConfig config = LogUnitServerConfig.parse(serverContext.getServerConfig());
+        StreamLog streamLog;
+        if (config.isMemoryMode()) {
+            log.warn("Log unit opened in-memory mode (Maximum size={}). "
+                    + "This should be run for testing purposes only. "
+                    + "If you exceed the maximum size of the unit, old entries will be "
+                    + "AUTOMATICALLY trimmed. "
+                    + "The unit WILL LOSE ALL DATA if it exits.", Utils
+                    .convertToByteStringRepresentation(config.getMaxCacheSize()));
+            streamLog = new InMemoryStreamLog();
+        }
+        else{
+            streamLog = new StreamLogFiles(serverContext, config.isNoVerify());
+        }
+
+        ImmutableMap<Class, AbstractServer> servers = ImmutableMap.<Class, AbstractServer>builder()
+                .put(BaseServer.class, new BaseServer(serverContext))
+                .put(SequencerServer.class, new SequencerServer(serverContext))
+                .put(LayoutServer.class, new LayoutServer(serverContext))
+                .put(LogUnitServer.class, new LogUnitServer(serverContext, streamLog))
+                .put(ManagementServer.class, new ManagementServer(serverContext, streamLog))
+                .build();
+
+        this.serverContext = serverContext;
```

Comment:
You can still continue calling the other constructor. Avoids duplicate and diverging code paths.

Reply:
Sounds good